NebulaPatcher/Patches/Dynamic/GalacticTransport_Patch.cs

```diff
@@ -0,0 +1,127 @@
+                Debug.Log("recycle " + __instance.stationRecycleCursor);
+                debug = __instance.stationRecycle[__instance.stationRecycleCursor-1];
+            }
+            else
+            {
+                Debug.Log("no recycle");
+                debug = __instance.stationCursor;
+            }
+            Debug.Log("adding station id " + station.id + " on " + GameMain.galaxy.PlanetById(planetId).displayName + " to gid " + debug);
+            /*
+            for (int i = 0; i < GameMain.data.galacticTransport.stationPool.Length; i++)
+            {
+                if(GameMain.data.galacticTransport.stationPool[i]?.id == station.id && GameMain.data.galacticTransport.stationPool[i]?.planetId == planetId)
+                {
+                    Debug.Log("replacing station " + station.id + " on " + GameMain.galaxy.PlanetById(planetId).displayName + " at gid " + station.gid + " with provided one from server");
+                    GameMain.data.galacticTransport.stationPool[i] = null;
+                    GameMain.data.galacticTransport.stationRecycle[GameMain.data.galacticTransport.stationRecycleCursor++] = i;
+                    return true;
+                }
+            }
+            // if we are a client and have no fake station then just add the new one normally
+            // this should happen when a player places an ILS on a FactoryData known to this client
+            // or when this client arrives at a planet for the first time which contains ILS
+            */
+            return true;
+        }
+
+        [HarmonyPostfix]
+        [HarmonyPatch("AddStationComponent")]
+        public static void AddStationComponent_Postfix(GalacticTransport __instance)
+        {
+            if(__instance.stationRecycleCursor > 0)
+            {
+                Debug.Log("CURSOR NOW AT: " + __instance.stationCursor + " recycle: " + __instance.stationRecycleCursor + "(" + __instance.stationRecycle[__instance.stationRecycleCursor - 1] + ")");
+            }
+            else
+            {
+                Debug.Log("CURSOR NOW AT: " + __instance.stationCursor + " recycle: " + __instance.stationRecycleCursor);
+            }
+            for(int i = 0; i < __instance.stationPool.Length; i++)
+            {
+                if(__instance.stationPool[i] != null)
+                {
+                    Debug.Log(i + ": " + __instance.stationPool[i].gid + " " + __instance.stationPool[i].id + " " + __instance.stationPool[i].planetId);
+                }
+                else
+                {
+                    Debug.Log(i + ": null");
+                }
+            }
+        }
+
+        [HarmonyPostfix]
+        [HarmonyPatch("RemoveStationComponent")]
+        public static void RemoveStationComponent_Postfix(GalacticTransport __instance, int gid)
+        {
+            if (__instance.stationRecycleCursor > 0)
```

Comment:
Shouldn't this be a message sent to the host to try and remove the station component?

Especially, since this should be reference counted per player, so you don't remove station components that are still in use?

Reply:
yes i probably should make this a message to the host too, thanks.

Reply:
This is removed in current code, i added an issue for the error popping up when a client removes an ILS (#241 )